packages/ld/ld.ts

```diff
@@ -8,62 +8,71 @@ export type LdOptions<T extends object> = {
 
-function proxyfy<T extends object>(obj: T, options: LdOptions<T>): T {
+const cachedTable: TableMap<unknown> = new Map()
+
+function proxify<T extends object>(obj: T, options: LdOptions<T>): T {
   if (undefined === options.root) {
-    table.clear()
+    cachedTable.clear()
     options.root = obj
   }
 
   return new Proxy<T>(obj, {
-    get: (target, prop, receiver) => {
-      const value = Reflect.get(target, prop, receiver);
+    get: (target: T, prop: string, receiver: unknown) => {
+      const value: string extends keyof T ? T[string] : unknown = Reflect.get(target, prop, receiver);
 
       if (typeof value === 'object' && value !== null) {
-        return proxyfy<any>(value, options);
+        return proxify<any>(value, options);
       }
 
-      if (typeof value !== 'string') {
-        return value
+      if (typeof value !== 'string' && typeof value !== 'number' && typeof value !== 'symbol') {
+        return value;
       }
 
+      const valueStr = value.toString();
+
       // TODO: pattern matcher
       let absoluteValue = undefined
-      if (!options.urlPattern.test(value)) {
+      if (!options.urlPattern.test(valueStr)) {
         if (options.relativeURIs === false) {
-          return value;
+          return valueStr;
         }
 
-        if (value[0] !== '/') {
-          return value;
+        if (valueStr[0] !== '/') {
+          return valueStr;
         }
 
-        absoluteValue = `${options.urlPattern.protocol}://${options.urlPattern.hostname}${value}`;
+        absoluteValue = `${options.urlPattern.protocol}://${options.urlPattern.hostname}${valueStr}`;
         if (!options.urlPattern.test(absoluteValue)) {
-          return value;
+          return valueStr;
         }
       }
 
-      if (table.has(value)) {
-        return table.get(value);
+      if (cachedTable.has(valueStr)) {
+        return cachedTable.get(valueStr);
       }
 
-      table.set(value, undefined);
-      ; ((iri, object, t, uri) => {
-        (options as LdOptions<T>).fetchFn(uri)
-          .then(data => {
-            t.set(iri, data)
-            if (options.onUpdate) {
-              options.onUpdate(object, { iri, data })
-            }
-          })
-          .catch((err) => options.onError(err))
-      })(value, proxyfy((options as LdOptions<T>).root, options), table, absoluteValue === undefined ? value : absoluteValue)
-      return table.get(value)
+      cachedTable.set(valueStr, undefined);
+
+      const callFetch = (iri: string, object: T, tableMap: TableMap<T>, uri: RequestInfo | URL) => {
+        options.fetchFn(uri)
+            .then(data => {
+              tableMap.set(iri, data)
+              if (options.onUpdate) {
+                options.onUpdate(object, {iri, data})
+              }
+            })
+            .catch((err) => options.onError(err))
+      };
+
+      callFetch(valueStr, proxify(options.root, options), cachedTable as TableMap<T>, absoluteValue === undefined ? valueStr : absoluteValue)
```

Comment:
what's wrong with the anonymous IIFE ? :D actually fine with that imo it's better for stack traces (ignore my comment)